File: src/main/java/com/primogemstudio/advancedui/render/Shaders.java
package com.primogemstudio.advancedui.render;

import com.mojang.blaze3d.vertex.DefaultVertexFormat;
import ladysnake.satin.api.managed.ManagedCoreShader;
import ladysnake.satin.api.managed.ManagedShaderEffect;
import ladysnake.satin.api.managed.ShaderEffectManager;
import net.minecraft.resources.ResourceLocation;

import static com.primogemstudio.advancedui.AdvancedUI.MOD_ID;

public class Shaders {
    public static final ManagedCoreShader ROUNDED_RECT = ShaderEffectManager.getInstance().manageCoreShader(new ResourceLocation(MOD_ID, "rounded_rect"), DefaultVertexFormat.POSITION_COLOR);
    public static final ManagedShaderEffect FAST_GAUSSIAN_BLUR = ShaderEffectManager.getInstance().manage(new ResourceLocation(MOD_ID, "shaders/filter/fast_gaussian_blur.json"));
    public static final ManagedShaderEffect GAUSSIAN_BLUR = ShaderEffectManager.getInstance().manage(new ResourceLocation(MOD_ID, "shaders/filter/gaussian_blur.json"));
    public static void init() {
    }
}

File: src/main/java/com/primogemstudio/advancedfmk/render/Shaders.kt
package com.primogemstudio.advancedfmk.render

import com.mojang.blaze3d.vertex.DefaultVertexFormat
import com.primogemstudio.advancedfmk.mmd.renderer.CustomRenderType
import ladysnake.satin.api.managed.ManagedCoreShader
import ladysnake.satin.api.managed.ManagedShaderEffect
import ladysnake.satin.api.managed.ShaderEffectManager
import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.ShaderInstance
import net.minecraft.resources.ResourceLocation


object Shaders {
    @JvmField
    val ROUNDED_RECT = ShaderInstance(
        Minecraft.getInstance().resourceManager, "rounded_rect",
        DefaultVertexFormat.POSITION_COLOR
    )
    @JvmField
    val ROUNDED_RECT_CLIP = ShaderInstance(
        Minecraft.getInstance().resourceManager, "rounded_rect_clip",
        DefaultVertexFormat.POSITION_COLOR
    )
    @JvmField
    val ROUNDED_RECT_TEX = ShaderInstance(
        Minecraft.getInstance().resourceManager, "rounded_rect_tex",
        DefaultVertexFormat.POSITION_COLOR_TEX
    )
    @JvmField
    val MMD_SHADER = ShaderInstance(
        Minecraft.getInstance().resourceManager, "mmd_entity",
        CustomRenderType.ENTITY
    )
    @JvmField
    val FAST_GAUSSIAN_BLUR: ManagedShaderEffect = ShaderEffectManager.getInstance()
        .manage(ResourceLocation("shaders/filter/fast_gaussian_blur.json"))
    @JvmField
    val GAUSSIAN_BLUR: ManagedShaderEffect = ShaderEffectManager.getInstance()
        .manage(ResourceLocation("shaders/filter/gaussian_blur.json"))
}

